oak_functions_service/src/lib.rs

```diff
@@ -145,15 +145,53 @@ impl schema::OakFunctions for OakFunctionsService {
 
     fn update_lookup_data(
         &mut self,
-        lookup_data: &schema::LookupData,
-    ) -> Result<schema::Empty, micro_rpc::Status> {
-        let data = lookup_data
+        request: &schema::UpdateLookupDataRequest,
+    ) -> Result<schema::UpdateLookupDataResponse, micro_rpc::Status> {
+        let update_action = map_update_action(request.action)?;
+
+        let data = request
+            .chunk
+            .as_ref()
+            .unwrap()
             .items
             .iter()
-            .map(|entry| Ok((entry.key.clone(), entry.value.clone())))
-            .collect::<Result<_, micro_rpc::Status>>()?;
+            .map(|entry| (entry.key.clone(), entry.value.clone()))
+            .collect();
+
+        let update_status = self.lookup_data_manager.update_data(update_action, data);
+
+        Ok(schema::UpdateLookupDataResponse {
+            update_status: map_update_status(update_status).into(),
+        })
+    }
+}
+
+// This crate depends on oak_functions_lookup and schema, so we cannot use schema in
+// oak_functions_lookup. As a work-around we define and map a corresponding type for UpdateAction.
+fn map_update_action(action: i32) -> Result<oak_functions_lookup::UpdateAction, micro_rpc::Status> {
+    let action =
+        schema::UpdateAction::from_i32(action).ok_or(micro_rpc::Status::new_with_message(
+            micro_rpc::StatusCode::InvalidArgument,
+            format!("could not map action from i32"),
+        ))?;
+
+    match action {
+        schema::UpdateAction::StartAndFinish => {
+            Ok(oak_functions_lookup::UpdateAction::StartAndFinish)
+        }
+        _ => Err(micro_rpc::Status::new_with_message(
+            micro_rpc::StatusCode::InvalidArgument,
+            format!("action not implemented: {:?}", action),
+        )),
+    }
+}
 
-        self.lookup_data_manager.update_data(data);
-        Ok(schema::Empty {})
+// This crate depends on oak_functions_lookup and schema, so we cannot use schema in
+// oak_functions_lookup. As a work-around we define and map a corresponding type for UpdateStatus.
+fn map_update_status(status: oak_functions_lookup::UpdateStatus) -> schema::UpdateStatus {
```

Comment:
Can this be implemented as Into or From?

Reply:
Done! Thanks---I had the wrong first thought that I can't, but I can define Into/From for `schema::*` here.